### PR TITLE
integrate categories page with mock server

### LINKED ISSUE #95: integrate mock server with categories page

### PR DESCRIPTION
**Pull request by amarchenko25**
closed #95

### CODE REVIEW

**Review by Platton96 — DISMISSED** [review]

File: ecommers/lib/core/services/membership_service.dart [hunk 1]
```diff
@@ -17,50 +18,45 @@ class MembershipService {
   bool get isNotExpired =>
       _expirationDate != null && DateTime.now().isBefore(_expirationDate);
 
-  MembershipService(this.secureStorage);
+  MembershipService();
+
+  Future initialize() async {
+    _storage = await SharedPreferences.getInstance();
+    load();
```
Comment (Platton96) [inline ecommers/lib/core/services/membership_service.dart]: I don't understand what load. I should find method.  Name is not informative

File: ecommers/lib/ui/widgets/category/sub_category_list.dart [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      builder: (_, provider, child) {
+        final category = provider.selectedCategory;
+        final subCategories = category.subCategories;
+
+        return SurfaceContainer(
+          child: ListView.separated(
+            shrinkWrap: true,
+            physics: const NeverScrollableScrollPhysics(),
+            itemCount: subCategories.length,
+            padding: const EdgeInsets.all(0.0),
```
Comment (Platton96) [inline ecommers/lib/ui/widgets/category/sub_category_list.dart:25]: maybe u could remove padding.

File: ecommers/lib/core/services/membership_service.dart (hunk 1 above)
Comment (amarchenko25) [reply ecommers/lib/core/services/membership_service.dart]: Anyway, if I rename it to loadData, you will find that method and try to understand what the data is

File: ecommers/lib/ui/widgets/category/sub_category_list.dart (hunk 2 above)
Comment (amarchenko25) [reply ecommers/lib/ui/widgets/category/sub_category_list.dart:25]: no, it fixes a bug related to list

File: ecommers/lib/ui/widgets/menu/menu_list.dart [hunk 3]
```diff
@@ -1,73 +1,52 @@
...
-              shrinkWrap: true,
-              physics: const NeverScrollableScrollPhysics(),
-              itemCount: itemList.length,
-              padding: const EdgeInsets.all(0.0),
-              itemBuilder: (context, index) {
-                final itemModel = itemList[index];
+            shrinkWrap: true,
+            physics: const NeverScrollableScrollPhysics(),
+            itemCount: itemList.length,
+            padding: const EdgeInsets.all(0.0),
```
Comment (Platton96) [inline ecommers/lib/ui/widgets/menu/menu_list.dart:29]: why do u use padding 0

Comment (amarchenko25) [reply ecommers/lib/ui/widgets/menu/menu_list.dart:29]: it fixes a bug related to list

File: ecommers/lib/core/services/membership_service.dart (hunk 1 above)
Comment (Platton96) [reply ecommers/lib/core/services/membership_service.dart]: loadDataFromStorage

Comment (Platton96) [reply ecommers/lib/core/services/membership_service.dart]: loadSessionData

**Review by Platton96 — APPROVED** [review]

**Pull request merged**